packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -570,6 +570,15 @@ export class SourceFile {
         try {
             //TODO: this isnt ideal because it re-reads the file for each cell which is unnecessary
             source = getIPythonCells(this.fileSystem, this.getRealUri(), this._console)?.[cellIndex]?.source;
+            // Make sure we don't return any non-python cells
+            if (
+                source &&
+                ['%sql', '%%sql', '%sh', '%pip', '%run', '%fs', '%load', '%matplotlib', '%who', '%env'].includes(
```

Comment:
i'm also a bit skeptical about this change because it prevents errors from correctly appearing for users that are just working with regular notebooks. we still want to report an error for this invalid syntax on regular notebooks.

i think if we do this we need a new setting called `databricksNotebooks` or something

Reply:
I think a more generic setting allowing to skip cells based on magic would be good